src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,357 @@
+    }
+
+    private void UpdateTooltips()
+    {
+        pickerButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_PICK_COLOUR");
+        addPresetButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_ADD_PRESET");
+        hsvCheckButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_HSV_BUTTON_TOOLTIP");
+        rawCheckButton.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_RAW_BUTTON_TOOLTIP");
+
+        if (HsvMode)
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_H_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_S_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_V_TOOLTIP");
+        }
+        else
+        {
+            sliderROrH.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_R_TOOLTIP");
+            sliderGOrS.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_G_TOOLTIP");
+            sliderBOrV.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_B_TOOLTIP");
+        }
+
+        sliderA.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_A_TOOLTIP");
+    }
+
+    private void UpdateButtonsState()
+    {
+        if (hsvCheckButton == null)
+            return;
+
+        hsvCheckButton.Pressed = HsvMode;
+        rawCheckButton.Pressed = RawMode;
+
+        hsvCheckButton.Disabled = !hsvButtonEnabled || RawMode;
+        rawCheckButton.Disabled = !rawButtonEnabled || HsvMode;
+    }
+
+    private void OnAddPresetButtonPressed()
+    {
+        AddPreset(Color);
+    }
+
+    private void OnPresetSelected(Color colour)
+    {
+        Color = colour;
+        EmitSignal("colour_changed", Color);
+    }
+
+    private void OnPresetDeleted(TweakedColourPickerPreset preset)
+    {
+        presets.Remove(preset);
+        presetsContainer.RemoveChild(preset);
+        base.ErasePreset(preset.Color);
+        preset.QueueFree();
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        HsvMode = isOn;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        RawMode = isOn;
+    }
+
+    private void OnColourChanged(Color colour)
+    {
+        htmlColourEdit.Text = colour.a8 == 255 ? colour.ToHtml(false) : colour.ToHtml(true);
+    }
+
+    /// <summary>
+    ///   Called when (keyboard) entered in HtmlColourEdit or from OnHtmlColorEditFocusExited.
+    ///   Set Color when text is valid; reset if not.
+    /// </summary>
+    /// <param name="colour">Current htmlColourEditor text</param>
+    private void OnHtmlColourChanged(string colour)
+    {
+        if (colour.IsValidHtmlColor())
+        {
+            Color = new Color(colour);
+        }
+        else
+        {
+            htmlColourEdit.Text = Color.ToHtml();
+        }
+    }
+
+    /// <summary>
+    ///   Called when focus exited HtmlColorEdit.
+    /// </summary>
+    private void OnHtmlColorEditFocusExited()
```

Comment:
Hhyyrylainen wouldn't like this name :)
```suggestion
    private void OnHtmlColourEditFocusExited()
```

Reply:
Hard to change language habit...